engine/src/main/java/org/terasology/identity/ClientIdentity.java

```diff
@@ -49,4 +50,21 @@ public PrivateIdentityCertificate getPlayerPrivateCertificate() {
         }
         return playerPrivateCertificate;
     }
+
+    @Override
+    public int hashCode() {
+        return playerPrivateCertificate.hashCode() ^ playerPrivateCertificate.hashCode();
```

Comment:
Wanted to XOR the public with the private but accidentally typed private two times. I just found there is an [Objects.hash](https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#hash-java.lang.Object...-) for this purpose of building hashcodes for objects composed by multiple fields, would that be OK? Will change again if you prefer the manual implementation.